src/plugins/auctionHouseModule/Bid.ts

```diff
@@ -0,0 +1,104 @@
+import { PublicKey } from '@solana/web3.js';
+import {
+  amount,
+  BigNumber,
+  DateTime,
+  lamports,
+  Pda,
+  SolAmount,
+  SplTokenAmount,
+  toBigNumber,
+  toDateTime,
+  toOptionDateTime,
+} from '@/types';
+import { BidReceiptAccount } from './accounts';
+import { TokenWithMetadata } from '../nftModule';
+import { assert, Option } from '@/utils';
+import { AuctionHouse } from './AuctionHouse';
+
+export type Bid = Readonly<{
+  model: 'bid';
+  lazy: false;
+
+  // Models.
+  auctionHouse: AuctionHouse;
+  token: TokenWithMetadata;
+
+  // Addresses.
+  tradeStateAddress: Pda;
+  buyerAddress: PublicKey;
+  bookkeeperAddress: Option<PublicKey>;
+  receiptAddress: Option<Pda>;
+  purchaseReceiptAddress: Option<PublicKey>;
+
+  // Data.
+  price: SolAmount | SplTokenAmount;
+  tokens: SplTokenAmount;
+  createdAt: DateTime;
+  canceledAt: Option<DateTime>;
+}>;
```

Comment:
This model only works with private bids. I would like the `Bid` model to also support public bids, i.e. we don't bid on a particular token account, we just bid on the mint account.

What do you think of something like that?

```suggestion
export type Bid = Readonly<{
  model: 'bid';
  lazy: false;

  // Models.
  auctionHouse: AuctionHouse;

  // Addresses.
  tradeStateAddress: Pda;
  buyerAddress: PublicKey;
  bookkeeperAddress: Option<PublicKey>;
  receiptAddress: Option<Pda>;
  purchaseReceiptAddress: Option<PublicKey>;

  // Data.
  price: SolAmount | SplTokenAmount;
  tokens: SplTokenAmount;
  createdAt: DateTime;
  canceledAt: Option<DateTime>;
} & (
  { 
    isPublic: false;
    token: TokenWithMetadata;
  } 
  | { 
    isPublic: true;
    mint: MintWithMetadata;
  }
)>;
```

Note that this means,
- the `LazyBid` model also needs to support public bids. It would omit `token` or `mint` but would still have `isPublic` based on if we have a token address or not.
- the `createBidOperation` needs to support that too and send the right instruction based on the input.